packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2540,23 +2969,56 @@ export class TransactionController extends BaseControllerV1<
     );
   }
 
-  private addPendingTransactionTrackerListeners() {
-    this.pendingTransactionTracker.hub.on(
+  private removeIncomingTransactionHelperListeners(
+    incomingTransactionHelper = this.incomingTransactionHelper,
+  ) {
+    incomingTransactionHelper.hub.removeAllListeners('transactions');
+    incomingTransactionHelper.hub.removeAllListeners(
+      'updatedLastFetchedBlockNumbers',
+    );
+  }
+
+  private addIncomingTransactionHelperListeners(
+    incomingTransactionHelper = this.incomingTransactionHelper,
+  ) {
+    incomingTransactionHelper.hub.on(
+      'transactions',
+      this.onIncomingTransactions.bind(this),
+    );
+    incomingTransactionHelper.hub.on(
+      'updatedLastFetchedBlockNumbers',
+      this.onUpdatedLastFetchedBlockNumbers.bind(this),
+    );
+  }
+
+  private removePendingTransactionTrackerListeners(
+    pendingTransactionTracker = this.pendingTransactionTracker,
+  ) {
+    pendingTransactionTracker.hub.removeAllListeners('transaction-confirmed');
```

Comment:
Would this be better encapsulated in the `PendingTransactionTracker` itself in a method like `removeAllListeners`?

Reply:
Bump.

Reply:
It seems to make sense for the caller of `PendingTransactionTracker.hub.on` to also be responsible for unsubscribing when it's done listening to those events. I think this is okay as is.